week-four/app/src/main/java/pro/marvinhosea/movielist/repository/UserSharedPrefRepository.kt

```diff
@@ -0,0 +1,72 @@
+package pro.marvinhosea.movielist.repository
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+
+object UserSharedPrefRepository : UserRepository {
+
+    private lateinit var applicationContext: Context
+    private const val USERNAME = "USERNAME"
+    private const val USER_PASSWORD = "USER_PASSWORD"
+    private const val USER_IS_LOGGED_IN = "USER_IS_LOGGED_IN"
+
+    private fun sharedPreps() = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+
+    fun init(context: Context) {
+        applicationContext = context
+    }
+
+    /**
+     * Register user on shared preference
+     */
+    override fun registerUser(username: String, password: String) {
+        val editor = sharedPreps().edit()
+
+        editor.putString(USERNAME, username).apply()
+        editor.putString(USER_PASSWORD, password).apply()
+        editor.putBoolean(USER_IS_LOGGED_IN, true).apply()
+    }
+
+    /**
+     * Login in user by checking user validity
+     */
+    override fun loginUser(username: String, password: String): Boolean {
+        val getUsername = sharedPreps().getString(USERNAME, "")
+        val getPassword = sharedPreps().getString(USER_PASSWORD, "")
+
+        if (getUsername == null || getPassword == null) {
```

Comment:
Rename `getUsername` & `getPassword` to `storedUsername` & `storedPassword` or just username and password.

I don't see the point in passing in the user/pass if you don't compare them to anything.

if(username == null || password == null || username.isEmpty() || password.isEmpty()) {}

else {
// "log in"
}